library_generation/utilities.sh

```diff
@@ -268,3 +273,58 @@ get_proto_path_from_preprocessed_sources() {
   popd > /dev/null # sources
   echo "${result}"
 }
+
+# for a pre-processed library stored in $preprocessed_sources_path, a folder
+# tree is built on $target_folder so it looks like a googleapis-gen folder and
+# is therefore consumable by an .OwlBot.yaml file
+build_owlbot_cli_source_folder() {
+  local postprocessing_target=$1
+  local target_folder=$2
+  local preprocessed_sources_path=$3
+  proto_path=$(get_proto_path_from_preprocessed_sources "${preprocessed_sources_path}")
+  owlbot_staging_folder="${postprocessing_target}/owl-bot-staging"
+  mkdir -p "${owlbot_staging_folder}"
+
+  # By default (thanks to generation templates), .OwlBot.yaml `deep-copy` section
+  # references a wildcard pattern matching a folder
+  # ending with `-java` at the leaf of proto_path. We then use a generated-java
+  # folder that will be picked up by copy-code
+  mkdir -p "${target_folder}/${proto_path}/generated-java"
+  copy_directory_if_exists "${preprocessed_sources_path}" "proto" \
+    "${target_folder}/${proto_path}/generated-java/proto-google-cloud-library"
+  copy_directory_if_exists "${preprocessed_sources_path}" "grpc" \
+    "${target_folder}/${proto_path}/generated-java/grpc-google-cloud-library"
+  copy_directory_if_exists "${preprocessed_sources_path}" "gapic" \
+    "${target_folder}/${proto_path}/generated-java/gapic-google-cloud-library"
+  copy_directory_if_exists "${preprocessed_sources_path}" "samples" \
+    "${target_folder}/${proto_path}/generated-java/samples"
+  pushd "${target_folder}"
+  # create an empty commit so owl-bot-copy can process this as a repo
```

Comment:
Why owl-bot-copy require the folder to be a Git repo?

Reply:
I believe it's related to the (not used by us) [logic](https://github.com/googleapis/repo-automation-bots/blob/d2bb2ee39f96b4fabc1844e1994cf29abd74d7fa/packages/owl-bot/src/copy-code.ts#L672-L676) of telling apart which commits are related to an affected library in googleapis-gen

There are several `git log` calls that prepare a PR or commit message

Reply:
I see. Now that we don't copy from googleapis-gen and don't generate a pr anymore, we probably don't have to initialize a git repo(at least not here, new-client.py probably still need it to create a PR). But since we are still relying on owlbot-cli for copying code, it seems we have to live with it for now. cc: @JoeWang1127 Let's see if we can move this logic to other scripts in a later milestone.